dev/bots/runner_utils.dart

```diff
@@ -0,0 +1,255 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math' as math;
+
+import 'package:file/file.dart' as fs;
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+
+import 'run_command.dart';
+import 'tool_subsharding.dart';
+import 'utils.dart';
+
+typedef ShardRunner = Future<void> Function();
+
+final String exe = Platform.isWindows ? '.exe' : '';
+final String bat = Platform.isWindows ? '.bat' : '';
+final String flutterRoot = path.dirname(path.dirname(path.dirname(path.fromUri(Platform.script))));
+final String flutter = path.join(flutterRoot, 'bin', 'flutter$bat');
+final String dart = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dart$exe');
+final String pubCache = path.join(flutterRoot, '.pub-cache');
+final String engineVersionFile = path.join(flutterRoot, 'bin', 'internal', 'engine.version');
+
+const String kShardKey = 'SHARD';
+const String kSubshardKey = 'SUBSHARD';
+
+final String luciBotId = Platform.environment['SWARMING_BOT_ID'] ?? '';
+final bool runningInDartHHHBot =
+    luciBotId.startsWith('luci-dart-') || luciBotId.startsWith('dart-tests-');
+
+const String CIRRUS_TASK_NAME = 'CIRRUS_TASK_NAME';
+
+/// Environment variables to override the local engine when running `pub test`,
+/// if such flags are provided to `test.dart`.
+final Map<String,String> localEngineEnv = <String, String>{};
+
+String? _shuffleSeed;
+
+String get shuffleSeed {
+  if (_shuffleSeed == null) {
+    // Attempt to load from the command-line argument
+    final String? seedArg = Platform.environment['--test-randomize-ordering-seed'];
+    if (seedArg != null) {
+      _shuffleSeed = seedArg;
+    } else {
+      // Fallback to the original time-based seed generation
+      final DateTime seedTime = DateTime.now().toUtc().subtract(const Duration(hours: 7));
+      _shuffleSeed = '${seedTime.year * 10000 + seedTime.month * 100 + seedTime.day}';
+    }
+  }
+  return _shuffleSeed!;
+}
+
+// TODO(sigmund): includeLocalEngineEnv should default to true. Currently we
+// only enable it on flutter-web test because some test suites do not work
+// properly when overriding the local engine (for example, because some platform
+// dependent targets are only built on some engines).
+// See https://github.com/flutter/flutter/issues/72368
+Future<void> runDartTest(String workingDirectory, {
```

Comment:
Should runFlutterTest (from https://github.com/flutter/flutter/pull/146210) also move into this file? What decides if something goes into runner_utils.dart or utils.dart?

Reply:
Yes we can include `runFlutterTest` with this change as well. 

It might make more sense to migrate these changes into the existing `utils.dart`. My thought was that it could be helpful to have logic for triggering scripts/running tests separate from the existing utils contents of printing helper methods, but they are still closely related.